ci/functional/node_checkout.sh

```diff
@@ -0,0 +1,263 @@
+    echo -e "Number of Numa Nodes: ${num_numa_nodes}\n"
+else
+    echo "[FAILED] Number of sockets is not equal to number of numa nodes"
+    exit $rc
+fi
+
+}
+
+
+# Verify valid IB state
+verify_ibstate(){
+
+output=`ibstatus | egrep -e state`
+num_active_ports=$(echo $output | awk -F': ACTIVE' 'NF{print NF-1}')
+if [ "$num_active_ports" -gt 0 ]; then
+    echo -e "Number of active ib ports: ${num_active_ports}\n"
+else
+    echo "[FAILED] No active ib port"
+    exit $rc
+fi
+
+}
+
+# Verify all the IB interfaces have IP addresses attached
+verify_ip_for_interfaces(){
+
+# get all available interfaces
+available_ifaces=`ls /sys/class/net`
+echo $available_ifaces
+iface_count=$(echo $available_ifaces | awk -F' ' 'NF{print NF-1}')
+echo "Total number of interfaces: ${iface_count}"
+
+output=`ip -4 -o a | cut -d ' ' -f 2,7 | cut -d '/' -f 1`
+echo $output
+set -f
+array=(${available_ifaces// / })
+declare -a iface_with_ip
+declare -a iface_without_ip
+
+for i in "${!array[@]}";do
+    if [[ $output =~ "${array[i]}" ]]; then
+        echo "IP available for interface: ${array[i]}"
+        iface_with_ip=(${iface_with_ip[@]} "${array[i]}")
+    else
+        echo "IP not available for interface: ${array[i]}"
+        iface_without_ip=(${iface_without_ip[@]} "${array[i]}")
+    fi
+done
+
+pattern='ib'
+if [[ "${iface_without_ip[@]}" =~ "${pattern}" ]]; then
+    echo "IB interfaces need to IP address set"
+    exit $rc
+fi
+echo -e "\n"
+
+}
+
+# For dual IB setup, verify that the settings for multi-rail are set
+check_for_multirail_settings(){
+
+check_for=("net.ipv4.conf.all.accept_local" "net.ipv4.conf.all.arp_ignore" "net.ipv4.conf.ib0.rp_filter" "net.ipv4.conf.ib1.rp_filter")
+
+for i in "${!check_for[@]}"; do
+    output=`sysctl "${check_for[i]}"`
+    echo $output
+    value=$(echo $output | awk -F' ' '{print $3}')
+    echo $value
+#    if [[ $value != 0 ]]; then    # remove this when running on a hw where multi-rail is set
+    if [[ $value == 0 ]]; then
+        echo "[FAILED] ${check_for[i]} not set"
+        exit $rc
+    fi
+done
+
+}
+
+# UNCOMMENT WHEN ntpstat IS INSTALLED
+# Verify that NTP is configured and clocks are in sync
+verify_ntp(){
+
+output=`ntpstat`
+if [[ $output =~ 'synchronised to NTP server' ]]; then
+    echo "NTP configuration is set"
+else
+    echo "[FAILED] Check NTP congiration"
+    exit $rc
+fi
+
+}
+
+# Check SSD health
+ssd_health(){
+
+# Are my nvme block devices present
+#ls /home/standan/daos.??*
+#daos_server storage prepare -p 4096 -u root --reset
+daos_server storage prepare --nvme-only -p 4096 -u root --reset
+sleep 2
+bash -c 'ls -l /dev/nvme*n1'
+local rc_ls_nvme=`echo $?`
+if [[ $rc_ls_nvme != 0 ]]; then
+    echo "[FAILED] No nvme block device found"
+    exit $rc
+fi
+
+# Identify the Ids of Nvme devices
+declare -a nvme_ids
+output=`lspci | grep Non-Volatile | awk '{print $1}'`
+local nvme_ids=(${output// / })
+for i in "${!nvme_ids[@]}"; do
+    echo "${nvme_ids[i]}"
```

Comment:
This seems an unnecessary indirection and can be fixed with:
```suggestion
for i in "${nvme_ids[@]}"; do
    echo "$i"
```